indico/modules/users/schemas.py

```diff
@@ -83,3 +106,360 @@ class Meta:
+
+
+class RoomExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = Room
+        fields = ('id', 'name', 'verbose_name', 'building', 'floor', 'number',
+                  'location_name', 'site', 'map_url')
+
+
+class StaticSiteExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = StaticSite
+        fields = ('id', 'event_id', 'event_url', 'state', 'requested_dt', 'url')
+
+    url = Function(lambda site: url_for('static_site.download', site, _external=True))
+    event_url = Function(lambda site: url_for('events.display', site.event, _external=True))
+
+
+class RegistrationTagExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = RegistrationTag
+        fields = ('title',)
+
+
+class RegistrationFileExportSchema(Schema):
+    class Meta:
+        fields = ('registration_id', 'field_data_id', 'filename', 'md5', 'url', 'path')
+
+    url = Function(lambda data: url_for('event_registration.registration_file', data.locator.file, _external=True))
+    path = Function(lambda file: build_storage_path(file))
+
+
+class RegistrationDataExportSchema(Schema):
+    title = Function(lambda data: data.field_data.field.title)
+    description = Function(lambda data: data.field_data.field.description)
+    data = Method('_get_data')
+
+    def _get_data(self, data):
+        friendly_data = data.field_data.field.get_friendly_data(data, for_humans=True)
+        return self._strip_lazy_strings(friendly_data)
+
+    def _strip_lazy_strings(self, data):
+        if isinstance(data, _LazyString):
+            return str(data)
+        elif isinstance(data, dict):
+            return {k: self._strip_lazy_strings(v) for k, v in data.items()}
+        elif isinstance(data, (list, tuple)):
+            return [self._strip_lazy_strings(v) for v in data]
+        else:
+            return data
+
+    @post_dump(pass_original=True)
+    def _add_file(self, data, original, **kwargs):
+        """For file fields, export also the file metadata"""
+        if original.filename:
+            file_data = RegistrationFileExportSchema().dump(original)
+            return data | {'file': file_data}
+
+
+class RegistrationExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = Registration
+        fields = ('id', 'url', 'event_id', 'event_title', 'event_url', 'state', 'currency', 'price', 'submitted_dt',
+                  'email', 'first_name', 'last_name', 'is_deleted', 'checked_in', 'checked_in_dt', 'rejection_reason',
+                  'consent_to_publish', 'created_by_manager', 'tags', 'data')
+
+    price = fields.Decimal(as_string=True)
+    url = Function(lambda reg: url_for('event_registration.registration_details', reg, _external=True))
+    event_title = Function(lambda reg: reg.event.title)
+    event_url = Function(lambda reg: url_for('events.display', reg.event, _external=True))
+    tags = List(Nested(RegistrationTagExportSchema))
```

Comment:
I don't think tags should be exported. Those are internal to the event organizers and I think they do not expect them to be disclosed to users.... We also don't export manager-only fields.